tests/lib/server.py

```diff
@@ -7,12 +7,13 @@
 from textwrap import dedent
 from unittest.mock import Mock
 
-from pip._vendor.contextlib2 import nullcontext
 from werkzeug.serving import WSGIRequestHandler
 from werkzeug.serving import make_server as _make_server
 
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
+from ..compat import nullcontext
```

Comment:
Probably better to put `nullcontext` in `test.lib.compat` instead, this fits existing code structuring better.

Reply:
That works for me. Done in the latest revision (and rebased).